compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -137,8 +120,8 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
                     validateEntityAnnotation(ctx, typeDefinitionNode, recordTypeSymbol);
                     validateRecordFieldsAnnotation(ctx, recordNode,
                             ((ModulePartNode) ctx.syntaxTree().rootNode()).members());
-                    validateRecordField(ctx, typeDefinitionNode, recordTypeSymbol, memberNodes);
-                    if ((hasPersistAnnotation || isPersistEntity)) {
+                    if (hasPersistAnnotation) {
+                        validateRecordFieldType(ctx, recordTypeSymbol.fieldDescriptors());
```

Comment:
Wouldnt it be easier to check this case in the else part of if clause in 123. This is much clearer. If you dont have persist annotation, we just have to loop through field with annotation and add error if they use other annotations.